lightly/api/api_workflow_upload_dataset.py

```diff
@@ -35,13 +35,26 @@
 from lightly.openapi_generated.swagger_client.rest import ApiException
 
 
+try:
+    from lightly.data import LightlyDataset
+    LightlyDatasetType = LightlyDataset
+    _lightly_dataset_available = True
+except (
+    ImportError,
+    OSError,
+    ModuleNotFoundError
+):
+    _lightly_dataset_available = False
+    LightlyDatasetType: str = "LightlyDataset"
+
+
 class _UploadDatasetMixin:
     """Mixin to upload datasets to the Lightly Api.
 
     """
 
     def upload_dataset(self,
-                       input: Union[str, LightlyDataset],
+                       input: Union[str, LightlyDatasetType],
```

Comment:
Can we not set it to `input: Union[str, "LightlyDataset"]`? Then the docs will be nicer as users don't have to figure out what `LightlyDatasetType` is.

Reply:
You are right, it's nicer without.